components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/dao/impl/ApplicationDAOImpl.java

```diff
@@ -3357,6 +3386,85 @@ private String resolveSQLFilter(String filter) {
         return sqlfilter;
     }
 
+    private FilterData getFilterDataForDBQuery(String filter) {
+
+        FilterData filterData = new FilterData();
+
+        if (StringUtils.isBlank(filter) || filter.equals(ASTERISK)) {
+            filterData.setFilterString("SP_APP.APP_NAME LIKE ?");
+            filterData.addFilterValue("%");
+        } else {
+            // Not repeating validations as all validations are handled in the Identity API Server.
+            try {
+                FilterTreeBuilder filterTreeBuilder = new FilterTreeBuilder(filter);
+                Node rootNode = filterTreeBuilder.buildTree();
+                if (rootNode instanceof ExpressionNode) {
+                    ExpressionNode expressionNode = (ExpressionNode) rootNode;
+
+                    filterData.setFilterString(generateFilterStringForBackend(expressionNode.getAttributeValue(),
+                            expressionNode.getOperation()));
+                    filterData.addFilterValue(generateFilterValueForBackend(expressionNode.getOperation(),
+                            expressionNode.getValue()));
+                } else {
+                    // Currently, supports only filters with one AND/OR operation.
+                    // Have to recursively traverse the filter tree to support more than one operation.
+                    OperationNode operationNode = (OperationNode) rootNode;
+                    ExpressionNode expressionLeftNode = (ExpressionNode) rootNode.getLeftNode();
+                    ExpressionNode expressionRightNode = (ExpressionNode) rootNode.getRightNode();
+
+                    String formattedLeftNodeString = generateFilterStringForBackend(
+                            expressionLeftNode.getAttributeValue(), expressionLeftNode.getOperation());
+                    String formattedRightNodeString = generateFilterStringForBackend(
+                            expressionRightNode.getAttributeValue(), expressionRightNode.getOperation());
+
+                    filterData.setFilterString(formattedLeftNodeString + " " +
+                            operationNode.getOperation() + " " + formattedRightNodeString);
+                    filterData.addFilterValue(generateFilterValueForBackend(expressionLeftNode.getOperation(),
+                            expressionLeftNode.getValue()));
+                    filterData.addFilterValue(generateFilterValueForBackend(expressionRightNode.getOperation(),
+                            expressionRightNode.getValue()));
+                }
+            } catch (IOException | IdentityException e) {
+                log.error("Error occurred while converting filter for backend.", e);
```

Comment:
why doesn't this error is thrown to upper levels? What's the API response if the request go into this catch?

Reply:
Resolved in https://github.com/wso2/carbon-identity-framework/pull/4432